File: remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int curr = nums[0];
        int seen = 1;
        int p = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] != curr) {
                nums[p] = nums[i];
                p++;
                seen = 1;
                curr = nums[i];
            } else if (seen < 2) {
                nums[p] = nums[i];
                p++;
                seen++;
            }
        }
        return p;
    }
}

File: 80-remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        p = 1
        seen = False
        for i in range(1, len(nums)):
            if nums[i] > nums[i - 1]:
                nums[p] = nums[i]
                p += 1
                seen = False
            elif seen == False:
                nums[p] = nums[i]
                seen = True
                p += 1
        return p